File: cpp/linked-list.cpp
#include <iostream>
using namespace std;

class student {
public: 
    int ID; // Member variable to store the student ID
    student *next; // Pointer to the next student in the list
    student() {
        ID = 0; // Initialize the ID to 0 and set next pointer to NULL
        next = NULL;
    }

    void setstudent(int id) {
        this->ID = id; // Set the student ID
        this->next = NULL; // Set the next pointer to NULL
    }

    int getstudent() {
        return this->ID; // Return the student ID
    }

    void nextelement(student *ns) {
        this->next = ns; // Set the next pointer to the given student pointer
    }

    student* getnextstudent() {
        return next; // Return the next student pointer
    }
};

class StudentList {
private: 
    student *HeadStudent, *CurrentStudent;
public:
    StudentList() : HeadStudent(NULL), CurrentStudent(NULL) {} // Initialize head and current pointers to NULL

    void Add_Student(int studentid) {
        student *NewStudent = new student(); // Create a new student node
        NewStudent->setstudent(studentid); // Set the new student ID
        if (HeadStudent == NULL) { // If the list is empty, set the head to the new student
            HeadStudent = NewStudent;
            CurrentStudent = NewStudent;
        } else { // Otherwise, add the new student after the current student
            CurrentStudent->nextelement(NewStudent);
            CurrentStudent = NewStudent; // Update the current student pointer to the new student
        }
        cout << "Student ID: " << CurrentStudent->getstudent() << endl;
    }
};

int main() {
    StudentList sl;
    sl.Add_Student(56);
    sl.Add_Student(78);
    sl.Add_Student(12);
    return 0;
}

File: python/linked-list.py
#add new element at the beginning of the linked-list

class Student:
    def __init__(self, student_id=0):
        self.id = student_id
        self.next = None

class StudentList:
    def __init__(self):
        self.head = None
        self.current = None
        self.tail = None
        
    def add_student(self, student_id):
        new_student = Student(student_id)
        if self.current is None:
            self.head = new_student
            self.current = new_student
            self.tail = new_student
        else: 
            self.current.next = new_student
            self.current = new_student
            self.tail = new_student
            
    def insert_beginning(self, student_id):
        new_student = Student(student_id)
        new_student.next = self.head
        self.head = new_student
        if self.current is None:
            self.current = new_student
        if self.tail is None:
            self.tail = new_student
            
    def print_list(self):
        temp = self.head
        while temp is not None:
            print(temp.id, end=' ')
            temp = temp.next    
        print()

if __name__ == "__main__":
    sl = StudentList()
    sl.add_student(56)
    sl.add_student(78)
    sl.add_student(12)
    sl.insert_beginning(99)
    sl.print_list()
